apps/coscrad-frontend/src/utils/generic-components/presenters/multiple-property-presenter.tsx

```diff
@@ -0,0 +1,31 @@
+import { SinglePropertyPresenter } from './single-property-presenter';
+
+export type PropertyLabels<T> = {
+    [K in keyof T]?: string;
+};
+
+interface MultiPropertyPresenterProps<T> {
+    keysAndLabels: PropertyLabels<T>;
+    data: T;
+}
+
+export const MultiplePropertyPresenter = <T,>({
+    keysAndLabels,
+    data,
+}: MultiPropertyPresenterProps<T>): JSX.Element => {
+    return (
+        <>
+            {Object.entries(keysAndLabels).map(([propertyKey, label]) => {
+                // `year` and `numberOfPages` in book are both numbers
```

Comment:
Not sure what the best way to do this is.

Reply:
The best way would be to map them to strings before passing them in. Remember what I said about `MultiplePropertyPresenter` doesn't know what the "real" view models are. You can do a mapping before you pass the `data` in. I'd rather not see you calling `toString` here.

Reply:
Note that while reference types in JavaScript will inherit `toString` from Object.prototype, and most primitive types' autobox wrapper has `toString` as well, this **will** blow up in case some how null or undefined come through. If you want a general way to turn unknown input into a string, use `JSON.stringify(...)`, but this is not good to show the user in general.